packages/wasm/crate/src/planner.rs

```diff
@@ -1,591 +1,389 @@
+        let swap_commitment = swap_commitment
+            .ok_or_else(|| anyhow!("missing swap commitment in swap claim"))?
+            .try_into()?;
+
+        let swap_record: SwapRecord = storage
+            .get_swap_by_commitment(swap_commitment)
+            .await?
+            .ok_or_else(|| anyhow!("Swap record not found"))?
+            .try_into()?;
+
+        let swap_claim = SwapClaimPlan {
+            swap_plaintext: swap_record.swap,
+            position: swap_record.position,
+            output_data: swap_record.output_data,
+            epoch_duration: sct_params.epoch_duration,
+            proof_blinding_r: Fq::rand(&mut OsRng),
+            proof_blinding_s: Fq::rand(&mut OsRng),
+        }
+        .into();
 
-    /// Submit a new governance proposal in this transaction.
-    pub fn proposal_submit(&mut self, proposal: Proposal, deposit_amount: Amount) -> &mut Self {
-        self.action(ActionPlan::ProposalSubmit(ProposalSubmit {
-            proposal,
-            deposit_amount,
-        }));
-        self
+        actions.push(swap_claim);
     }
 
-    /// Withdraw a governance proposal in this transaction.
-    pub fn proposal_withdraw(&mut self, proposal: u64, reason: String) -> &mut Self {
-        self.action(ActionPlan::ProposalWithdraw(ProposalWithdraw {
-            proposal,
-            reason,
-        }));
-        self
+    for tpr::Delegate { amount, rate_data } in request.delegations {
+        let amount = amount
+            .ok_or_else(|| anyhow!("missing amount in delegation"))?
+            .try_into()?;
+        let rate_data: RateData = rate_data
+            .ok_or_else(|| anyhow!("missing rate data in delegation"))?
+            .try_into()?;
+        actions.push(rate_data.build_delegate(amount).into());
     }
 
-    /// Claim a governance proposal deposit in this transaction.
-    pub fn proposal_deposit_claim(
-        &mut self,
-        proposal: u64,
-        deposit_amount: Amount,
-        outcome: ProposalOutcome<()>,
-    ) -> &mut Self {
-        self.action(ActionPlan::ProposalDepositClaim(ProposalDepositClaim {
-            proposal,
-            deposit_amount,
-            outcome,
-        }));
-        self
+    for tpr::Undelegate { value, rate_data } in request.undelegations {
+        let value: Value = value
+            .ok_or_else(|| anyhow!("missing value in undelegation"))?
+            .try_into()?;
+        let rate_data: RateData = rate_data
+            .ok_or_else(|| anyhow!("missing rate data in undelegation"))?
+            .try_into()?;
+        actions.push(rate_data.build_undelegate(value.amount).into());
     }
 
-    /// Cast a validator vote in this transaction.
-    pub fn validator_vote(&mut self, vote: ValidatorVote) -> &mut Self {
-        self.action(ActionPlan::ValidatorVote(vote));
-        self
+    /*
+    for tpr::UndelegateClaim { .. } in request.undelegation_claims {
+        // need to wait for a new release of monorepo
```

Comment:
We need to wait for the web repo to consume a new release of the core Rust crates to pick up these proto definitions. But when we do, this will be the place to process them.